### PR TITLE
Feature/1343 dynamic topper species icon

### PR DESCRIPTION
**Pull request by zamuzakki**
This is PR for #1343 
[Kazam_screencast_00047.webm](https://github.com/kartoza/sawps/assets/7352963/5c198983-4dbb-4e2b-85af-82592c4f97b9)

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 38 sec

File: django_project/species/models.py [hunk 1]
```diff
@@ -57,6 +83,39 @@ class Meta:
         verbose_name_plural = "Taxa"
         db_table = "taxon"
 
+    def save(self, *args, **kwargs):
+        print(self.graph_icon)
```
Comment (dimasciput) [inline django_project/species/models.py:87]: @zamuzakki please remove this

File: django_project/species/models.py [hunk 2]
```diff
@@ -57,6 +83,39 @@ class Meta:
...
+        if self.graph_icon:
+            graph_icon = self.graph_icon.read()
+            topper_icon = graph_icon.replace(
```
Comment (dimasciput) [inline django_project/species/models.py:90]: Please update the code to use regex instead. Users should also be able to upload non-black svg files. Use this function :

```
def replace_fill_color(data, new_color):
    fill_pattern = re.compile(br'fill="[^"]*"', re.IGNORECASE)
    return fill_pattern.sub(br'fill="' + new_color.encode() + br'"', data)
```

**Review by dimasciput — CHANGES_REQUESTED** [review]

File: django_project/species/models.py (hunk 1 above)
Comment (zamuzakki) [reply django_project/species/models.py]: I've removed this

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 9 min 15 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 13 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 12 min

**Review by dimasciput — APPROVED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 5 min 28 sec

**Pull request merged**